### ISSUE TITLE
cmd/go: when 'go build' run in directory with go.mod and no .go files: "cannot find module for path ."

### ISSUE THREAD

**Issue by GingerMoon** [golang/go#35414]
### What version of Go are you using (`go version`)?

<pre>
go version go1.12.2 linux/amd64

</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"

</pre></details>

### What did you do?

I want to put several common modules into one repo.
So the repo layout looks like below:
```
common-modules/
        |-- module1/
                |-- *.go files
        |-- module2/
                |-- *.go files
        go.mod
```
The go.mod file looks like:
```
module company1.com/common-modules

go 1.12
```
Run "go build" under directory common-modules.

### What did you expect to see?
No error is returned.

### What did you see instead?
If I run "go build" under directory common-modules, error returns:
can't load package: package company1.com/common-modules: unknown import path "company1.com/common-modules": cannot find module providing package company1.com/common-modules

**Comment by ianlancetaylor** [comment]
CC @bcmills @jayconrod

**Comment by bcmills** [comment]
> go version go1.12.2

I believe we fixed some related bugs in 1.13. What happens if you try this with the current release (Go 1.13.4)?

**Comment by GingerMoon** [comment]
Hi @bcmills 
The result is the same  for 1.13.4:

[myname@mycompany myproject]$ ll
total 4
drwxrwxr-x. 5 myname myname 100 Nov 11 10:40 fe3
-rw-rw-r--. 1 myname myname  47 Nov 11 10:40 go.mod
drwxrwxr-x. 6 myname myname 107 Nov  7 20:04 ll3
[myname@mycompany myproject]$ go build
build .: cannot find module for path .
[myname@mycompany myproject]$ cat go.mod 
module git.mycompany.com/cloud/myproject

go 1.13
[myname@mycompany myproject]$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/myname/.cache/go-build"
GOENV="/home/myname/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/myname/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/gorootpath/go-1.13.4"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/gorootpath/go-1.13.4/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/myname/myproject/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build343362267=/tmp/go-build -gno-record-gcc-switches"
[myname@mycompany myproject]$ go build
build .: cannot find module for path .
